components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/tcp/core/TCPConfiguration.java

```diff
@@ -0,0 +1,195 @@
+/*
+ *
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * /
+ */
+
+package org.wso2.carbon.inbound.endpoint.protocol.tcp.core;
+
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * loading the tcp.properties file from repository/conf
+ */
+public class TCPConfiguration {
+    private static final Logger log = Logger.getLogger(TCPConfiguration.class);
+
+    private static String CONF_LOCATION = "conf.location";
+    private static TCPConfiguration instance = new TCPConfiguration();
+    private Properties props;
+
+    private TCPConfiguration() {
+        try {
+            props = loadProperties("tcp.properties");
+        } catch (Exception exception) {
+            log.info("could not load the properties from tcp.properties file...");
```

Comment:
Would there be an exception here, since most of possible exception are already caught in loadProperties method. If any, better to catch specific exception and include error in log message. Also it should be an error log, not info.